rust/benches/benchmarks/fifo_window.rs

```diff
@@ -0,0 +1,56 @@
+include!("../../tests/common.rs");
+
+use criterion::*;
+// use swag::reactive::Reactive;
+use criterion_cpu_time::PosixTime;
+use swag::recalc::ReCalc;
+use swag::soe::SoE;
+use swag::two_stacks::TwoStacks;
+use swag::FifoWindow;
+
+criterion_group! {
+    name = experiments;
+    config = Criterion::default()
+        .with_measurement(PosixTime::UserTime);
+    targets = suite
+}
+
+fn synthesize(size: u64) -> impl Iterator<Item = Int> {
+    (0..size as i64).map(Int)
+}
+
+fn run<BinOp, Window>(group: &mut BenchmarkGroup<PosixTime>, size_exp: u32, name: &str)
+where
+    Window: FifoWindow<Int, BinOp>,
+    BinOp: Operator,
+{
+    let window = &mut Window::new();
+    let group = group.throughput(Throughput::Elements(1));
+    let size = (2 as u64).pow(size_exp);
+    let data = synthesize(size);
+    // Prefill the window
+    data.for_each(|item| window.push(item));
+    window.query();
+    // Begin the benchmark
+    group.bench_function(BenchmarkId::new(name, size_exp), |bench| {
+        bench.iter(|| {
+            Window::push(black_box(window), black_box(Int(0)));
```

Comment:
It's probably best to not always push `0` as the new element. We did `i % 101`, where `i` was the iteration count of the total iterations in the experiment. The values are arbitrary, but I wanted to make sure there was some mix of values without also hitting any overflows.